docs/rfc/rfc-017-abci++-vote-extension-propag.md

```diff
@@ -553,6 +554,32 @@ Then, we need to handle two new situations, roughly equivalent to cases (h.1) an
   However we can manually have the node crash and recover as a workaround. This effectively converts
   this case into (h.1).
 
+### Upgrade Path
+
+v0.36 of Tendermint will be the first version to implement the vote extension system.
+The v0.36 release aims to be backwards compatible with the previous release, v0.35,
+which had no implementation of vote extensions and no mechanisms in place to facilitate
+their later addition.
+
+Vote extensions pose an issue for Tendermint upgrades. Chains that upgrade from
+v0.35 to v0.36 will attempt to produce the first height running v0.36 without vote
+extension data from the previous height. We intend to allow chains to _require_
+vote extensions data. Chains that do so will not make progress without vote
+extension data and the corresponding application will expect vote extension
+data to be present for [PrepareProposal](https://github.com/tendermint/tendermint/blob/cec0a9798/proto/tendermint/abci/types.proto#L129) call it receives. 
+
+To facilitate the upgrade and provide applications a mechanism to require
+vote extensions, we will provide application developers with a [ConsensusParam](https://github.com/tendermint/tendermint/blob/cec0a9798/proto/tendermint/types/params.proto#L13)
+to transition the chain from no history of vote extensions to requiring vote extensions.
+This parameter will be an `int64` representing the first height where vote extensions
+will be required for votes to be considered valid. Once the configured height occurs,
+the parameter will be disallowed from changing, meaning that vote extensions cannot flip from being
+required to again being optional. Whatever system is created for transitioning a
+chain to requiring vote extensions needs to save the height at which the
+parameter changed so that Tendermint can determine whether or not to enforce
+vote extension presence on the `PrepareProposal` call. For further clarification
+and implementation, see [issue 8453](https://github.com/tendermint/tendermint/issues/8453)
+
```

Comment:
I would like to have a modified version of:
(a) Solution 3 
(b) the base solution we have implemented

Regarding (b), #8453 explains it very clearly and concisely (consider bringing it here). As for (a), as I stated [here](https://github.com/tendermint/tendermint/issues/8453#issuecomment-1115823287), I'd be willing to take this on (on a further PR to this RFC).

Reply:
> I would like to have a modified version of:
(a) Solution 3
(b) the base solution we have implemented
 
Would you mind clarifying what you mean by this? In this document you'd like this written?

Reply:
If I am not mistaken, the upgrade mechanism will "break" both Solution 3 and the base solution described here and implemented by #8433... as they are today.
So IMO we need to explain how we are going to modify those to make them work.

For (b) I would suggest nothing really involved, apart from bringing the conditions you specified in #8453, a few words on how this will change the solution would do. Probably best to do when you have a clear idea on the code.
A priori, I can think of;
* the condition to switch from blocksync to consensus might have to change
* in blocksync, when we check if a "received block" message is valid, we need to extend the current check (now we could receive no extension in some conditions and the message would still be valid)
* how we verify in blocksync might have to be adapted as well (not sure)
* and I'm probably missing stuff

For (a), I can do it later (when I'm done with the spec cleanup).

What do you think?